src/main.rs

```diff
@@ -421,6 +423,63 @@ fn rotate() -> Option<u64> {
     }
 }
 
+#[derive(Deserialize, PartialEq, Message)]
+struct SubnetListRecord {
+    #[prost(bytes = "vec", repeated, tag = "2")]
+    pub subnets: ::prost::alloc::vec::Vec<::prost::alloc::vec::Vec<u8>>,
+}
+
+#[derive(Deserialize, Message)]
+struct SubnetRecord {
+    #[prost(bytes = "vec", repeated, tag = "3")]
+    pub membership: ::prost::alloc::vec::Vec<::prost::alloc::vec::Vec<u8>>,
+}
+
+#[derive(Clone, PartialEq, Message)]
+struct RegistryGetValueRequest {
+    #[prost(message, optional, tag = "1")]
+    pub version: ::core::option::Option<u64>,
+    #[prost(bytes = "vec", tag = "2")]
+    pub key: ::prost::alloc::vec::Vec<u8>,
+}
+
+#[derive(Clone, PartialEq, Message)]
+struct RegistryGetValueResponse {
+    #[prost(bytes = "vec", tag = "3")]
+    pub value: ::prost::alloc::vec::Vec<u8>,
+}
+
+pub fn make_get_value_request(
+    key: &str,
+) -> Vec<u8> {
+    let request = RegistryGetValueRequest { key: key.as_bytes().to_vec(), version: None };
+    let mut buf = Vec::new();
+    request.encode(&mut buf).unwrap();
+    buf
+}
+
+async fn update_permissions_from_registry() {
+    let registry_canister = Principal::from_slice(&vec![0]);
```

Comment:
I guess we'll need a different input here? :)

Reply:
https://rwlgt-iiaaa-aaaaa-aaaaa-cai.raw.ic0.app is what I was told by NNS team